gammapy/datasets/fermi.py

```diff
@@ -216,6 +218,135 @@ def fetch_fermi_diffuse_background_model(filename='gll_iem_v02.fit'):
+        self.unc_exp_index = self.cat_row['Unc_Exp_Index']
+        self.signif = self.cat_row['Signif_Avg']
+
+    def plot_spectrum(self, ax=None):
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+
+
+        bin_edges1 = np.zeros(0)
+
+        bin_edges2 = np.zeros(0)
+
+        x_vals = np.zeros(0)
+
+        y_vals = np.zeros(0)
+
+        y_upper = np.zeros(0)
+        y_lower = np.zeros(0)
+
+
+        for i in range(0, np.size(self.x_bins) - 1):
+
+            flux = self.cat_row[self.y_labels[i]]
+
+
+
+            # Require both a detection and a lower bound
+            if np.isnan(flux) == False:
+                y_err_label = "Unc_" + self.y_labels[i]
+
+                print(self.cat_row[y_err_label])
+                flux_lower_bound = self.cat_row[y_err_label][0]
+                if np.isnan(flux_lower_bound) == False:
+
+                    y_vals = np.append(y_vals, flux)
+
+                    #print(y_err_label)
+                    #print(self.cat_row[y_err_label])
+
+                    y_lower = np.append(y_lower, flux + self.cat_row[y_err_label][0])
+
+                    y_upper = np.append(y_upper, flux + self.cat_row[y_err_label][1])
+
+                    y_vals[-1] = np.log10(y_vals[-1])
+
+                    y_upper[-1] = np.log10(y_upper[-1])
+
+                    y_lower[-1] = np.log10(y_lower[-1])
+
+                    # y_val - new_y_lower = y_lower
+
+                    y_lower[-1] = -(y_lower[-1] - y_vals[-1])
+
+                    # y_val + new_y_upper = y_upper
+
+                    y_upper[-1] = y_upper[-1]  - y_vals[-1]
+
+                    x_vals = np.append(x_vals,
+                                       0.5 * (self.x_bins[i + 1] - self.x_bins[i]) + self.x_bins[i])
+
+                    #print(self.x_bins[i])
+                    #print(self.x_bins[i+1])
+                    #print(x_vals[i])
+
+                    # x_vals - bin_edge1 = x_bin[i]
+                    # x_vals + bin_edge2 = x_bin[i + 1]
+                    bin_edges1 = np.append(bin_edges1,
+                                           -(self.x_bins[i] - x_vals[-1]))
+
+                    bin_edges2 = np.append(bin_edges2,
+                                           self.x_bins[i + 1] - x_vals[-1])
+
+                    #print("")
+                    #print(x_vals[i] - bin_edges1[i])
+                    #print(x_vals[i] + bin_edges2[i])
+                    #raw_input(" ")
+
+
+
+        ax.errorbar(x_vals, y_vals,
+                    xerr=(bin_edges1, bin_edges2),
+                    yerr=(y_lower, y_upper),
+                    elinewidth=1, linewidth=0, color='black')
+        
+	#x_spectrum = np.linspace(np.log10(min(x_vals)), np.log10(max(x_vals)), 25)
+	#x_spectrum = 10**x_spectrum
+        if self.spec_type == "PowerLaw":
+            print(type(self.spec_index))
+            y_spectrum = self.int_flux * (10 ** x_vals / self.pivot_en) ** -self.spec_index
```

Comment:
Typically log-log plots are done in matplotlib using `plt.loglog()`, which scales the axes and not the data.